### PR TITLE
IGNITE-14798 Exclude the test phase when maven.test.skip is set true

### PR DESCRIPTION
**Pull request by SammyVimes**

### CODE REVIEW

File: parent/pom.xml [hunk 1]
```diff
@@ -449,6 +449,7 @@
             Profile to exclude running surefire (unit) tests but do not prevent running integration ones
         -->
         <profile>
+            <id>skip-surefire</id>
```
Comment (vveider) [inline parent/pom.xml:452]: Let's rename it as `skip-unit-tests` to comply with overall naming.

File: parent/pom.xml [hunk 2]
```diff
@@ -467,6 +468,53 @@
             </build>
         </profile>
 
+        <!--
+            Profile that activates automatically when maven.test.skip is set to true and excludes the test phase
```
Comment (vveider) [inline parent/pom.xml:472]: Comment is a bit confusing, let's rephrase:
```
Profile to exclude the test phase from maven lifecycle when maven.test.skip is set. It is required to mitigate the issue with test-scope dependencies in non tested builds
```
No need for additional info about automatic activation (previous profile has no such info too).

File: parent/pom.xml [hunk 3]
```diff
@@ -467,6 +468,53 @@
...
+        -->
+        <profile>
+            <id>skip-all-tests</id>
+            <activation>
+                <property>
+                    <name>maven.test.skip</name>
+                    <value>true</value>
```
Comment (vveider) [inline parent/pom.xml:479]: Value is unnecessary, because with use `-Dmaven.test.skip` notation in command line calls.

**Review by vveider — CHANGES_REQUESTED** [review]

File: parent/pom.xml (hunk 3 above)
Comment (SammyVimes) [reply parent/pom.xml:479]: I'd prefer to keep this explicit as it is so in maven's documentation

**Comment by ibessonov** [comment]
Has already been merged some time ago

**Pull request closed without merging**